src/module/system/check/check.ts

```diff
@@ -457,26 +460,48 @@ class CheckPF2e {
 
         let rerollFlavor = game.i18n.localize(`PF2E.RerollMenu.MessageKeep.${keep}`);
 
-        if (heroPoint) {
+        if (heroPoint && !usePoint) {
+            fu.logCompatibilityWarning('The heroPoint option is deprecated. Use usePoint: "hero" instead.', {
+                since: "7.3",
+                until: "7.4",
+            });
+            usePoint = "hero";
+        }
+
+        if (usePoint) {
             const rerollingActor = actor.isOfType("familiar") ? actor.master : actor;
 
-            // If the reroll costs a hero point, first check if the actor has one to spare and spend it
+            // If the reroll costs a hero or mythic point, first check if the actor has one to spare and spend it
             if (rerollingActor?.isOfType("character")) {
-                const heroPointCount = rerollingActor.heroPoints.value;
-                if (heroPointCount) {
-                    await rerollingActor.update({
-                        "system.resources.heroPoints.value": Math.clamp(
-                            heroPointCount - 1,
-                            0,
-                            rerollingActor.heroPoints.max,
-                        ),
-                    });
-                    rerollFlavor = game.i18n.format("PF2E.RerollMenu.MessageHeroPoint", { name: rerollingActor.name });
-                } else {
-                    ui.notifications.warn(
-                        game.i18n.format("PF2E.RerollMenu.WarnNoHeroPoint", { name: rerollingActor.name }),
-                    );
-                    return;
+                const resource = (() => {
+                    switch (usePoint) {
+                        case "hero":
+                            return {
+                                data: rerollingActor.heroPoints,
+                                name: "heroPoints",
+                                noPoints: "PF2E.RerollMenu.WarnNoHeroPoint",
+                                rerollFlavor: "PF2E.RerollMenu.MessageHeroPoint",
+                            };
+                        case "mythic":
+                            return {
+                                data: rerollingActor.system.resources.mythicPoints,
+                                name: "mythicPoints",
+                                noPoints: "PF2E.RerollMenu.WarnNoMythicPoint",
+                                rerollFlavor: "PF2E.RerollMenu.MessageMythicPoint",
+                            };
+                        default:
+                            return null;
+                    }
+                })();
+                if (resource) {
+                    const count = resource.data.value;
+                    if (count > 0) {
+                        await rerollingActor.updateResource(resource.name, Math.clamp(count - 1, 0, resource.data.max));
```

Comment:
Between the count > 0 and updateResource() already clamping, I don't think you need to clamp again.